### PR TITLE
emacs: add options to install different icons

### PR DESCRIPTION
**Pull request by d12frosted**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

---

Inspired by emacs-mac formula. New available options are:

- with-spacemacs-icon
- with-modern-icon

---

Didn't see any notes about new options for existing formula in Contributing guidelines, so hope that this commit will work for you :)

### CODE REVIEW

File: Library/Formula/emacs.rb [hunk 1]
```diff
@@ -98,6 +100,19 @@ def install
...
+      # icons
+      icon_file = "nextstep/Emacs.app/Contents/Resources/Emacs.icns"
+      modern_icons = "https://s3.amazonaws.com/emacs-mac-port/Emacs.icns.modern"
+      spacemacs_icons = "https://github.com/nashamri/spacemacs-logo/blob/master/spacemacs.icns?raw=true"
+      if build.with? "modern-icon"
+        rm "#{icon_file}"
+        curl "#{modern_icons}", "-o", "#{icon_file}"
+      elsif build.with? "spacemacs-icon"
+        rm "#{icon_file}"
+        curl "-L", "#{spacemacs_icons}", "-o", "#{icon_file}"
```
Comment (bfontaine) [inline Library/Formula/emacs.rb]: Use  `resource do` blocks instead of manually retrieving the file.

**Comment by bfontaine** [comment]
We try to limit the number of options in each formula; How common are these needs?

File: Library/Formula/emacs.rb (hunk 1 above)
Comment (d12frosted) [inline Library/Formula/emacs.rb]: Will do once I return to my computer.

**Comment by d12frosted** [comment]
Well, as for `modern_icon` - not sure. But Spacemacs icon will be used by pretty huge amount of users.

**Comment by d12frosted** [comment]
Looks like my commit has broken something. Will check a little bit later.

**Comment by dunn** [comment]
The formula already has a lot of options; I'm not sure I'm in favor of adding more unless they offer functionality needed by a large number of users.

**Comment by mikemcquaid** [comment]
Passing on this but thanks for the pull request!

**Comment by fintelkai** [comment]
Disappointing decision by @mikemcquaid. Why make Spacemacs users do this manually? Isn't this what homebrew is for?

**Comment by mikemcquaid** [comment]
@fintelkai It's not sustainable to keep adding more and more niche options; they aren't tested by our CI, they aren't supported by upstream and they will, as a result, eventually break. In short: allowing formulae to be customised in every way you'd like is not what Homebrew is for but the building blocks are still there to allow you to do it yourself (e.g. consider [creating your own tap](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md) instead. Thanks!

**Comment by d12frosted** [comment]
Separate tap for Spacemacs is also an option. Thanks.

**Pull request closed without merging**